lib/puppet-lint/plugins/check_unsafe_interpolations.rb

```diff
@@ -100,8 +100,8 @@ def get_exec_titles
       # Check if title is double quotes string
       elsif tokens[token_idx].next_code_token.next_code_token.type == :DQPRE
         # Find the start and end of the title
-        title_start_idx = tokens.rindex { |r| r.type == :DQPRE }
-        title_end_idx = tokens.rindex { |r| r.type == :DQPOST }
+        title_start_idx = tokens.find_index(tokens[token_idx].next_code_token.next_code_token)
+        title_end_idx = title_start_idx + colon_index_offset(tokens.slice(title_start_idx, tokens.length), ':')
```

Comment:
I wonder if we should use [a..n] notation over slice here? Only because we are already using it on L106.

Reply:
ahh good idea. I agree it would be neater if it can serve the same purpose. Will give it a go.